src/prognostic_equations/advection.jl

```diff
@@ -132,37 +138,41 @@ NVTX.@annotate function explicit_vertical_advection_tendency!(Yₜ, Y, p, t)
                     ᶠinterp(ᶜρʲs.:($$j)[colidx] - Y.c.ρ[colidx]) *
                     ᶠgradᵥ_ᶜΦ[colidx]
                 ) / ᶠinterp(ᶜρʲs.:($$j)[colidx])
+
+            # buoyancy term in mse equation
+            @. Yₜ.c.sgsʲs.:($$j).mse[colidx] +=
+                adjoint(CT3(ᶜinterp(Y.f.sgsʲs.:($$j).u₃[colidx]))) *
+                (ᶜρʲs.:($$j)[colidx] - Y.c.ρ[colidx]) *
+                ᶜgradᵥ(ᶠΦ[colidx]) / ᶜρʲs.:($$j)[colidx]
         end
 
         # TODO: Move this to implicit_vertical_advection_tendency!.
-        if p.atmos.turbconv_model isa PrognosticEDMFX
-            for j in 1:n
-                @. ᶜa_scalar[colidx] =
-                    draft_area(Y.c.sgsʲs.:($$j).ρa[colidx], ᶜρʲs.:($$j)[colidx])
-                vertical_transport!(
-                    Yₜ.c.sgsʲs.:($j).ρa[colidx],
-                    ᶜJ[colidx],
-                    ᶜρʲs.:($j)[colidx],
-                    ᶠu³ʲs.:($j)[colidx],
-                    ᶜa_scalar[colidx],
-                    dt,
-                    edmfx_upwinding,
-                )
-
-                vertical_advection!(
-                    Yₜ.c.sgsʲs.:($j).h_tot[colidx],
-                    ᶠu³ʲs.:($j)[colidx],
-                    Y.c.sgsʲs.:($j).h_tot[colidx],
-                    edmfx_upwinding,
-                )
-
-                vertical_advection!(
-                    Yₜ.c.sgsʲs.:($j).q_tot[colidx],
-                    ᶠu³ʲs.:($j)[colidx],
-                    Y.c.sgsʲs.:($j).q_tot[colidx],
-                    edmfx_upwinding,
-                )
-            end
+        for j in 1:n
```

Comment:
The if-end block is not needed as n is zero if turbconv_model is not PrognosticEDMFX